### ISSUE TITLE
BUG: code review bypass via commits to local branches

### ISSUE THREAD

**Issue by laurentsimon** [ossf/scorecard#1783]
Steps:
1. Dependabot publishes a change.
2. Maintainer edits that change to do whatever they want
3. Maintainer approves it
4. Maintainer merges it.

Let's try to verify that the approver != from all commits in the PR. In the config, branch protection is not enabled thru `*`, I think.

**Comment by laurentsimon** [comment]
another bypass I found: 
1. Create a PR
2. Click merge

The PR is considered reviewed.

Scorecard has this code https://github.com/ossf/scorecard/blob/main/checks/evaluation/code_review.go#L138 that accepts these because the committer is always "github". graphql has a field to know who clicked "auto-merge" this PR.

**Comment by laurentsimon** [comment]
> For this, can we do something like - expand the `PullRequest` struct to also contain commits (today we only care about merge commit)? 

yep that's also what I had in mind. Seems doable; it may increase rate limiting usage, but we'll see.

**Comment by azeemshaikh38** [comment]
Yeah, good thing is we have tests that will tell us by how much the API usage will go up. Would be good to get this fixed soon. Do you want to take this on or keep this open for someone else?

**Comment by azeemshaikh38** [comment]
> If nobody else wants it I'll take it, but if someone is interested I'm happy for them to take it.

I can take this on to reduce some of your load.

**Comment by laurentsimon** [comment]
SG, it's a fun (and important) one!

**Comment by azeemshaikh38** [comment]
Just an update - I'm working on this in 2 parts. (1) expose PR commit data through `CodeReviewRawData` struct. (2) figure out how to evaluate this data to find malicious activity while not causing false positives.

(1) is easy and I'm working on a PR for it. I'm still unsure about (2) and need to look into it more. Ideas are welcome.
